### PR TITLE
MRVA: Generate markdown for raw result tables

### PR DESCRIPTION
**Pull request by shati-patel**
Follow-up to #1318 for rendering raw results! This PR adds the actual table with results (see internal linked issue).

Some example rendered results [here](https://github.com/github/vscode-codeql/blob/shati-patel/md-raw-results/extensions/ql-vscode/test/pure-tests/remote-queries/markdown-generation/data/raw-results/results-repo1.md) and [here](https://github.com/github/vscode-codeql/blob/shati-patel/md-raw-results/extensions/ql-vscode/test/pure-tests/remote-queries/markdown-generation/data/raw-results/results-repo2.md) 🖼️ 🎨 

(It looks kinda ugly, but maybe that's mainly because these example results and column names are fairly meaningless 😂)

## Checklist

N/A - still internal only 🕳️ 

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts [hunk 1]
```diff
@@ -198,6 +202,52 @@
...
+      break;
+    case 'object':
+      {
+        const url = tryGetRemoteLocation(value.url, fileLinkPrefix);
+        cellValue = `[\`${convertNonPrintableChars(value.label)}\`](${url})`;
+      }
+      break;
+  }
+  // `|` characters break the table, so we need to escape them
+  return cellValue.replace('|', '\\|');
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts:247]: Ah...thanks code scanning! I think you want to use `replaceAll` here.

[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts:247]: Can you add a line to one of the tests where there are multiple `|` in the same table row?

[the remark this answers is not in the document]
Comment (shati-patel) [reply extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts:247]: > Ah...thanks code scanning! I think you want to use `replaceAll` here.

I tried that, but then I got this mysterious error 🤨 
image

Good idea about updating the tests! Will do that.

[the remark this answers is not in the document]
Comment (shati-patel) [reply extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts]: Ugh, I broke things more 😅 Will come back to this tomorrow!

[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts]: `replaceAll` was introduced in node 15. And the package.json requires node >=14.17.1. So, I'm guessing that your local node version is too old.

Either:

1. We change the requirement to node 15 or greater (double check _exactly_ which version the function was added)
2. Do multi-replacement in a different way. It should work if you call `replace` with a regex and use a `global` tag. eg- `/\\|/g` (this regex may not be right...., but it's something like that)

**Comment by charisk** [comment]
Give me a shout if you need any help getting things green!

**Review by charisk — APPROVED** [review]
LGTM!

**Pull request merged**